core/src/commonMain/kotlin/Peripheral.kt

```diff
@@ -241,4 +241,17 @@ internal suspend inline fun <reified T : State> Peripheral.suspendUntilOrThrow()
         .first { it is T }
 }
 
-internal expect val Peripheral.identifier: String
+public expect class Identifier
+
+/**
+ * The identifier of the remote device
+ *
+ * This identifier is the identifier returned by the operating system for the address of the device.
+ * The value of the identifier will differ between operating system implementations.  On iOS
+ * the identifier is a UUID.  On Android the identifier is a MAC address.  For security reasons
+ * the identifier is not a MAC address on iOS and may be a randomized MAC on Android.  The identifier
+ * may also change.
+ */
+public expect val Peripheral.identifier: Identifier
+
+public expect fun String.toIdentifier(): Identifier
```

Comment:
Being that the functionality of `Identifier` grew, I think it makes sense, for organizational purposes, to move this (and the corresponding platform implementations) to `Identifier.kt` files.